hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -89,34 +89,57 @@ spec = parallel $ do
                 failAfter 1 $ atomically (replicateM 1 (readTQueue queue2)) `shouldReturn` [arbitraryMsg]
                 failAfter 1 $ atomically (tryReadTQueue queue1) `shouldReturn` Nothing
 
-  prop "echoes history (past outputs) to client upon reconnection" $ \msgs -> monadicIO $ do
-    monitor $ cover 100 (null msgs) "no message when reconnecting"
-    monitor $ cover 100 (length msgs == 1) "only one message when reconnecting"
-    monitor $ cover 100 (length msgs > 1) "more than one message when reconnecting"
-    run . failAfter 5 $ do
-      withFreePort $ \port -> do
-        withAPIServer @SimpleTx "127.0.0.1" (fromIntegral port) alice mockPersistence nullTracer noop $ \Server{sendOutput} -> do
-          mapM_ sendOutput (msgs :: [ServerOutput SimpleTx])
-          withClient port $ \conn -> do
-            received <- replicateM (length msgs + 1) (receiveData conn)
-            case traverse Aeson.eitherDecode received of
-              Right msgs' -> msgs' `shouldBe` greeting : msgs
-              Left{} -> failure $ "Failed to decode messages " <> show msgs
+  it "echoes history (past outputs) to client upon reconnection" $
+    checkCoverage . monadicIO $ do
+      outputs <- pick arbitrary
+      monitor $ cover 0.1 (null outputs) "no message when reconnecting"
+      monitor $ cover 0.1 (length outputs == 1) "only one message when reconnecting"
+      monitor $ cover 1 (length outputs > 1) "more than one message when reconnecting"
```

Comment:
why did we change the cover values?

Reply:
Because they were wrong. 100% coverage on all cases was failing as soon as you add `checkCoverage`